### PR TITLE
Add tests for model attribute comparison in couch-to-sql toolchain

### PR DESCRIPTION
**Pull request by AmitPhulera**
## Technical Summary

While writing tests for Repeaters, I have raised a PR that would check if the couch model and SQL model have the same attributes.  @orangejenny suggested having them in the `couch-to-sql` toolchain and autogenerate the test file. Based on the feedback, I have updated it to include the test also at the end of `evaluate_couch_model_for_sql` command.

## Feature Flag

NA

### Safety story

Tested the command with `ReportConfiguration` model in local and it was working fine. And the changes does not effect any live feature of HQ and is just meant for couch-to-sql migration code generation.
### QA Plan

NA

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: docs/couch_to_sql_models.rst [hunk 1]
```diff
@@ -72,7 +72,13 @@ This should contain:
     - Some docs have attributes that are couch ids of other docs. These are weak spots easy to forget when the referenced doc type is migrated. Add a comment so these show up in a grep for the referenced doc type.
 
   - Run ``makemigrations``
-
+  - Add the test that was generated to it's respective place.
+    - The test file uses `ModelAttrEquality` util which has mehods for running the equality tests.
+    - The test class that is generated will have three methods namely `_couch_only_attrs`,  `_sql_only_attrs` and `test_have_same_attrs`.
+    - Generally during a migration some attributes and methods are renamed or removed as per need. To accomodate the changes you can update `_couch_only_attrs` and `_sql_only_attrs`.
+    - `_couch_only_attrs` is supposed to return a set of attributes and methods which are either removed, renamed or not used anymore in SQL.
+    - `_sql_only_attrs` would return a set of attributes and methods that are added newly in the SQL model.
```
Comment (millerdev) [inline docs/couch_to_sql_models.rst:80]:
```suggestion
    - `_sql_only_attrs` would return a set of attributes and methods that are new in the SQL model.
```

File: docs/couch_to_sql_models.rst [hunk 2]
```diff
@@ -72,7 +72,13 @@ This should contain:
...
+    - `test_have_same_attrs` is the test method that would run to test the equality of the attributes. The default implementation should work if you have populated above methods but you can modify it's implementation as per your requirement.
```
Comment (millerdev) [inline docs/couch_to_sql_models.rst:81]:
```suggestion
    - `test_have_same_attrs` will test the equality of the attributes. The default implementation should work if you have populated the above methods but you can modify it's implementation as needed.
```

File: corehq/apps/cleanup/management/commands/templates/migration_attr_equality_test.j2 [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    def _couch_only_attrs(cls):
+        return {
+            # removed attrs
+            # renamed attrs
+            # not used attrs
+        }
+
+    @classmethod
+    def _sql_only_attrs(cls):
+        return {}
```
Comment (millerdev) [inline corehq/apps/cleanup/management/commands/templates/migration_attr_equality_test.j2:23]: Do these need to be methods? Would it work if they were class attributes?
```suggestion
    couch_only_attrs = {
        # removed attrs
        # renamed attrs
        # not used attrs
    }
    sql_only_attrs = {}
```
This seems like a reasonable default, and they can always be changed to methods decorated with `django.utils.functional.classproperty` if a particular migration needs to do a more complex calculation.

Also, implied in the above example, what do you think of having them not be "private" (remove leading `_`)?

Obviously these suggestions require adjustments in the code that references these attributes.

Comment (AmitPhulera) [reply corehq/apps/cleanup/management/commands/templates/migration_attr_equality_test.j2]: Updated the related code and documentation as per the changes.

**Review by millerdev — APPROVED** [review]

**Pull request merged**